cmd/corald/service/service.go

```diff
@@ -53,3 +54,26 @@ func RewritePath(c *web.Context, targetPath string) func(*http.Request) {
 
 	return f
 }
+
+func requestService(c *web.Context, verb string, url string, payload []byte) (*http.Response, error) {
```

Comment:
You can use `request.Client.New` here instead of creating this function. Lets get this standardized to use `internal/platform/request`. You could create a package level global as:

```go
var DefaultSpongeServiceClient request.Client

...

func init() {
	if err := cfg.Init(cfg.EnvProvider{Namespace: ...}); err != nil {
		log.Println("Unable to initialize configuration")
		os.Exit(1)
	}

	// Insert the base url for requests by this client.
	DefaultSpongeServiceClient.BaseURL = ... // TODO: set to the base url of the client

	platformPrivateKey, err := ... // TODO: set to the secret/certificate for the downstream
	if err != nil || platformPrivateKey == "" {
		if err != nil {
			log.Printf("Downstream Auth : Disabled : %s\n", err.Error())
			return
		}

		log.Printf("Downstream Auth : Disabled\n")
		return
	}

	// If the platformPrivateKey is provided, then we should generate the token
	// signing function to be used when composing requests down to the platform.
	signer, err := auth.NewSigner(platformPrivateKey)
	if err != nil {
		log.Printf("Downstream Auth : Error : %s", err.Error())
		os.Exit(1)
	}

	// Requests can now be signed with the given signer function which we will
	// save on the application wide context. In the event that a function
	// requires a call down to a downstream platform, we will include a signed
	// header using the signer function here.
	DefaultSpongeServiceClient.Signer = signer

        ...
}
```

The above was partially sourced from https://github.com/coralproject/shelf/blob/master/cmd/sponge/web/web.go#L24